.github/workflows/ci-trivy-container-scan.yaml

```diff
@@ -42,7 +42,42 @@ jobs:
           - 'latest'
     env:
       IMAGE_REF: '${{ matrix.docker-image }}:${{ matrix.docker-tag }}'
+      JDK_DISTRIBUTION: corretto
+      CI_JDK_MAJOR_VERSION: 21
+
     steps:
+      - name: checkout
+        uses: actions/checkout@v4
+
+      - name: Cache Maven dependencies
+        uses: actions/cache@v4
+        timeout-minutes: 5
+        with:
+          path: |
+            ~/.m2/repository/*/*/*
+            !~/.m2/repository/org/apache/pulsar
+          key: ${{ runner.os }}-m2-dependencies-core-modules-${{ hashFiles('**/pom.xml') }}
+          restore-keys: |
+            ${{ runner.os }}-m2-dependencies-core-modules-
+
+      - name: Set up JDK
+        uses: actions/setup-java@v4
+        with:
+          distribution: ${{ env.JDK_DISTRIBUTION }}
+          java-version: ${{ env.CI_JDK_MAJOR_VERSION }}
+
+      - name: restore maven build results from Github artifact cache
+        run: |
+          cd $HOME
+          $GITHUB_WORKSPACE/build/pulsar_ci_tool.sh restore_tar_from_github_actions_artifacts pulsar-maven-repository-binaries
```

Comment:
this works only within pulsar-ci.yaml. 

we could consider adding the trivy scan for the master branch as part of pulsar-ci.yaml since there's also a scheduled build using it. The job could be made to run only for the scheduled build.

Reply:
Originally I was going to write a script that used the Trivy CLI and then this workflow would invoke that script which would allow easy execution elsewhere (devs running it ad-hoc, build running it, etc..). For the sake of simplicity and time I went w/ the GH Trivy action. 

> [!NOTE]
> The Trivy action is just a lightweight convenience wrapper around the Trivy CLI and we can easily extract the command by inspecting the action [output](https://github.com/apache/pulsar/actions/runs/8478690174/job/23231442717#step:4:18)).

Reply:
@lhotari @onobc I moved it to the main workflow, so we can take advantage of the image that was already just built. I've changed to only run it in the daily builds instead of for each PR validation.